boards/nucleo-wl55jc/include/periph_conf.h

```diff
@@ -84,6 +84,13 @@ static const uart_conf_t uart_config[] = {
  */
 static const spi_conf_t spi_config[] = {
     {
+        .dev      = SUBGHZSPI, /* Internally connected to Sub-GHz radio Modem  */
+        .rccmask  = RCC_APB3ENR_SUBGHZSPIEN,
+        .apbbus   = APB3,
+    }
```

Comment:
@akshaim Murdock is complaining that fielsd are unitialized:

```
../../boards/nucleo-wl55jc/include/periph_conf.h:107:1: error: missing initializer for member 'spi_conf_t::miso_pin' [-Werror=missing-field-initializers]
../../boards/nucleo-wl55jc/include/periph_conf.h:107:1: error: missing initializer for member 'spi_conf_t::sclk_pin' [-Werror=missing-field-initializers]
../../boards/nucleo-wl55jc/include/periph_conf.h:107:1: error: missing initializer for member 'spi_conf_t::cs_pin' [-Werror=missing-field-initializers]
../../boards/nucleo-wl55jc/include/periph_conf.h:107:1: error: missing initializer for member 'spi_conf_t::mosi_af' [-Werror=missing-field-initializers]
../../boards/nucleo-wl55jc/include/periph_conf.h:107:1: error: missing initializer for member 'spi_conf_t::miso_af' [-Werror=missing-field-initializers]
../../boards/nucleo-wl55jc/include/periph_conf.h:107:1: error: missing initializer for member 'spi_conf_t::sclk_af' [-Werror=missing-field-initializers]
../../boards/nucleo-wl55jc/include/periph_conf.h:107:1: error: missing initializer for member 'spi_conf_t::cs_af' [-Werror=missing-field-initializers]
```

```suggestion
        .dev      = SUBGHZSPI, /* Internally connected to Sub-GHz radio Modem  */
        .rccmask  = RCC_APB3ENR_SUBGHZSPIEN,
        .apbbus   = APB3,
    }
```

Use `GPIO_UNDEF` for the GPIO's,, maybe it makes sense to add a `GPIO_AF_UNDEF` to `gpio_af_t` for this platform. @jia200x do you have another idea?

Reply:
I did exactly the same last night, Murdock passed by hard fault in CPU
```
    {
        .dev      = SUBGHZSPI, /* Internally connected to Sub-GHz radio Modem  */
        .mosi_pin = GPIO_UNDEF,
        .miso_pin = GPIO_UNDEF,
        .sclk_pin = GPIO_UNDEF,
        .cs_pin   = GPIO_UNDEF,
        .mosi_af  = GPIO_AF0,
        .miso_af  = GPIO_AF0,
        .sclk_af  = GPIO_AF0,
        .cs_af    = GPIO_AF0,
        .rccmask  = RCC_APB3ENR_SUBGHZSPIEN,
        .apbbus   = APB3,
    }
```

Reply:
the SPI init function must be modified :/ It crashes with GPIO_UNDEF

+1 to add GPIO_AF_UNDEF

Reply:
I just pushed the changes and a check for valid gpio in spi ( #16625 ).  If GPIO is not used then GPIO_AF_UNDEF  would not really make a difference right?

Reply:
> I just pushed the changes and a check for valid gpio in spi ( #16625 ). If GPIO is not used then `GPIO_AF_UNDEF ` would not really make a difference right?

It wouldn't for initialization but I think its cleaner, can you apply this patch:

```diff
diff --git a/cpu/stm32/include/periph_cpu.h b/cpu/stm32/include/periph_cpu.h
index a3caf8be85..2936dcdc1c 100644
--- a/cpu/stm32/include/periph_cpu.h
+++ b/cpu/stm32/include/periph_cpu.h
@@ -326,6 +326,8 @@ typedef enum {
     GPIO_AF14,              /**< use alternate function 14 */
     GPIO_AF15               /**< use alternate function 15 */
 #endif
+    GPIO_AF_UNDEF,          /** an UNDEF value definition, e.g. for register
+                                based spi */
 #endif
 } gpio_af_t;
 ```
 
 You can include it in this PR IMO but in a separate commit, `git commit -m "cpu/stm32: add GPUI_AF_UNDEF"`, and then use the definition to initialize those fields plese.